core/src/apps/management/recovery_device/recover.py

```diff
@@ -78,5 +78,5 @@ def process_slip39(words: str) -> Tuple[Optional[bytes], int, int]:
     else:
         mnemonics = storage.recovery_shares.fetch()
 
-    identifier, iteration_exponent, secret = slip39.combine_mnemonics(mnemonics)
+    identifier, iteration_exponent, secret, _ = slip39.combine_mnemonics(mnemonics)
     return secret, share.group_index, share.index
```

Comment:
we could return group_count here too. We could then use it in homescreen.py [here](https://github.com/trezor/trezor-firmware/blob/master/core/src/apps/management/recovery_device/homescreen.py#L183) instead of fetching group_count from storage.

Reply:
Ok, that could be done, but I need to rebase on top of the fix we've introduced in #486. Will do.

Reply:
As discussed IRL. I'll merge now and then have a look on the `_request_secret` where I fix this.